sympy/solvers/solvers.py

```diff
@@ -1699,7 +1700,7 @@ def _expand(p):
                 else:
                     soln = list(soln.keys())
 
-                if soln is not None:
+                if soln is not None and not type(gens[0]) == floor:
```

Comment:
This seems like a fragile method for handling this case. Why in particular should the type not be floor?

Reply:
I will check and improve this code.

Reply:
@oscarbenjamin On speculation, we can solve the polynomial when the generator is a ```floor``` but then we would have to check if the ```floor(x)``` is an integer or not. If it is not, then an empty solution will have to be returned. Your thoughts?